### PR TITLE
feat: update expression build helpers

### PR DESCRIPTION
**Pull request by crash7**
Implements a set of expressions helper to help building update expressions that supports simple put operations and also list appends.

```
const expression = compose(
      updateExpression(
        'SET',
        compose(
          put('message', message),
          append('myList', [
            {
              item: 1
            }
          ])
        )
      ),
      updateExpression(
        'REMOVE',
        remove('messageOLD'),
      )
    );
```

Code can be improved with a more pure-fp-ramda approach, will see in the future.

image

### CODE REVIEW

File: src/update-expression.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+ * Edit an existing item's attributes, or add a new item to a table if it does not already exist.
+ * You can put or add attribute values.
+ * @module Helpers
+ */
+
+const { ifElse, endsWith, compose, evolve, concat, dropLast, merge, flip } = require('ramda');
+
+/**
+ * @private
+ * because original concat sucks
```
Comment (nfantone) [inline src/update-expression.js]: 👏

File: src/update-expression.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+/**
+ * Returns a function that, when invoked with a set of default values for a dynamo expression,
+ * it creates the mandatory UpdateExpression, AttributeNames and AttributeValues needed to build
+ * the `action` operations created by the given `operation` function.
+ *
+ * @function
+ * @param {String} action to execute in dynamodb, it can be SET, ADD, DELETE or REMOVE
+ * @param {Function} operations that returns an object with the operations to execute and their parameters
+ */
+const updateExpression = (action, operations) =>
```
Comment (nfantone) [inline src/update-expression.js]: @crash7 If these functions are going to be _exported_ (i.e.: part of the API), mind turning them into actual named `function` with a `return` statement? Let's help debugging.

File: src/update-expression.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+      UpdateExpression: catcon(` ${action} `)
+    })
+  );
+
+/**
+ * Returns a function that creates the mandatory `UpdateExpression`, `AttributeNames` and `AttributeValues` needed to build
+ * a `put` operation and merges everything with the given object containing `UpdateExpression`, `AttributeNames`
+ * and `AttributeValues` keys
+ *
+ * @function
```
Comment (nfantone) [inline src/update-expression.js]: @crash7 Could you please include a `@see` link to the relevant Dynamo documentation for this operation? Same goes to all other related functions.

File: src/update-expression.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+ * @param {String} attribute
+ * @param {String} value
```
Comment (nfantone) [inline src/update-expression.js]: @crash7 Incomplete jsdoc.

File: src/update-expression.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    }),
+    ExpressionAttributeValues: merge({
+      [`:${attribute}`]: value
+    })
+  });
+
+/**
+ * Returns a function that creates the mandatory `UpdateExpression`, `AttributeNames` and `AttributeValues` needed to build
+ * a `remove` operation and merges everything with the given object containing `UpdateExpression`, `AttributeNames`
+ * and `AttributeValues` keys
```
Comment (nfantone) [inline src/update-expression.js]: @crash7 Missing a full stop (`.`).

File: src/update-expression.js [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+ *
+ * @function
+ * @param {String} attribute
```
Comment (nfantone) [inline src/update-expression.js]: @crash7 Incomplete jsdoc (missing `@param` definition and `@returns`).

File: src/update-expression.test.js [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+        ExpressionAttributeValues: {}
+      })
+    ).toEqual({
+      UpdateExpression: ' SET myCustom = :myValue',
+      ExpressionAttributeNames: { custom: 'custom' },
+      ExpressionAttributeValues: { custom: 'values' }
+    });
+  });
+
+  test('should omit the expression generation when the operations return empty expressions', () => {
```
Comment (nfantone) [inline src/update-expression.test.js]: @crash7 I'd like to see tests for when the `action` is: `null`, `undefined`, empty (`''`), not one of the expected values (`SET`, `REMOVE`, etc.).

Comment (nfantone) [reply src/update-expression.test.js]: Same goes to values for the available helpers.

File: src/update.test.js [hunk 8]
```diff
@@ -59,6 +59,21 @@ describe('the update function', () => {
     );
   });
 
+  test('should run the helper to generate `UpdateExpression` if the second argument it is a function', async () => {
```
Comment (nfantone) [inline src/update.test.js]: @crash7 Redundant `it` here: "if the second argument ~it~ is a function".

**Comment by crash7** [comment]
@nfantone should be ok now, ended moving the builders to a folder at the same level of `src` because doing
```
const createFlynamo = require('@flybondi/flynamo');
const { put } = require('@flybondi/flynamo/builders');
```

is so much cooler than
```
const createFlynamo = require('@flybondi/flynamo');
const { builders: { put } } = require('@flybondi/flynamo');
```

**Pull request merged**